scripts/main/tabindex.js

```diff
@@ -2,92 +2,142 @@
 
 	// iterate over all elements and set tabindex to stored value (i.e. make is not focussable)
-	tmp.each(function (i, e) {
-		// TODO: shorter notation
-		a = $(e).data("tabindex-saved");
-		$(e).attr("tabindex", a);
-	});
+	tmp.each(
+		/**
+		 * @param {number} i - the index
+		 * @param {Element} e - the HTML element
+		 * @this {Element} - identical to `e`
+		 */
+		function (i, e) {
+			// TODO: shorter notation
+			// TODO @Hallenser: What do you intended by the TODO `short notation`? Moreover: Why do we use `this` and `e`? They refer to the identical instance of a HTML element.
+			const a = $(e).data("tabindex-saved");
+			$(e).attr("tabindex", a);
+		}
+	);
 };
 
+/**
+ * @param {jQuery} elem
+ * @param {boolean} [saveFocusElement=false]
+ * @returns {void}
+ */
 tabindex.makeUnfocusable = function (elem, saveFocusElement = false) {
 	if (!lychee.enable_tabindex) return;
 
-	// Todo: Make shorter noation
+	// Todo: Make shorter notation
 	// Get all elements which have a tabindex
-	let tmp = $(elem).find("[tabindex]");
+	const tmp = elem.find("[tabindex]");
 
 	// iterate over all elements and set tabindex to -1 (i.e. make is not focussable)
-	tmp.each(function (i, e) {
-		$(e).attr("tabindex", "-1");
-		// Save which element had focus before we make it unfocusable
-		if (saveFocusElement && $(e).is(":focus")) {
-			$(e).data("tabindex-focus", true);
-			// Remove focus
-			$(e).blur();
+	tmp.each(
+		/**
+		 * @param {number} i - the index
+		 * @param {Element} e - the HTML element
+		 */
+		function (i, e) {
+			$(e).attr("tabindex", "-1");
+			// Save which element had focus before we make it unfocusable
+			if (saveFocusElement && $(e).is(":focus")) {
+				$(e).data("tabindex-focus", true);
+				// Remove focus
+				$(e).blur();
+			}
 		}
-	});
+	);
 
 	// Disable input fields
-	$(elem).find("input").attr("disabled", "disabled");
+	elem.find("input").attr("disabled", "disabled");
 };
 
+/**
+ * @param {jQuery} elem
+ * @param {boolean} [restoreFocusElement=false]
+ * @returns {void}
+ */
 tabindex.makeFocusable = function (elem, restoreFocusElement = false) {
 	if (!lychee.enable_tabindex) return;
 
-	// Todo: Make shorter noation
+	// Todo: Make shorter notation
 	// Get all elements which have a tabindex
-	let tmp = $(elem).find("[data-tabindex]");
-
-	// iterate over all elements and set tabindex to stored value (i.e. make is not focussable)
-	tmp.each(function (i, e) {
-		$(e).attr("tabindex", $(e).data("tabindex"));
-		// restore focus elemente if wanted
-		if (restoreFocusElement) {
-			if ($(e).data("tabindex-focus") && lychee.active_focus_on_page_load) {
-				$(e).focus();
-				$(e).removeData("tabindex-focus");
+	const tmp = elem.find("[data-tabindex]");
+
+	// iterate over all elements and set tabindex to stored value (i.e. make is not focusable)
+	tmp.each(
+		/**
+		 * @param {number} i
+		 * @param {Element} e
+		 */
+		function (i, e) {
+			$(e).attr("tabindex", $(e).data("tabindex"));
```

Comment:
Similar strangeness here. Where is the code that stores values in `data("tabindex")`? Clearly `tabindex.makeUnfocusable` doesn't do that; it just sets the attribute to `-1`...

I just can't help feeling that this code simply doesn't work... Maybe we should just add some comments to that effect and, as they say, let a sleeping dog lie...

Reply:
I assume it is in `build.js` but I am not sure at all.

The main problem is that `build.js` is some kind of "god file" which builds HTML code for completely unrelated aspects of the view (photo, album, sidebar, menu) and that other code, like this one at hand, relies upon `build.js` does particular things a certain way.

Reply:
Yeah, it feels like in some distant past _all_ the HTML code generation may have been in `build.js`. Over time though, it spread all over the codebase of course. `view.js` is another monster file like that with everything and the kitchen sink in there... Files like `album.js` make a lot more sense to me...

Reply:
Can we resolve this conversation for now? I assume you don't want to fix `build.js` in the PR, do you?